src/share/cxx/prim_driver.cpp

```diff
@@ -44,37 +47,35 @@ void prim_run_subcycle_c (const Real& dt, int& nstep, int& nm1, int& n0, int& np
 
   // Check if needed to compute diagnostics or energy
   bool compute_diagnostics = false;
-  bool compute_energy      = params.energy_fixer;
-  if (nstep_end%params.state_frequency==0 || nstep_end==tl.nstep0) {
-#ifndef NDEBUG
-    std::cout << "WARNING! You need to implement something at line " << __LINE__ << " of file " << __FILE__ << "\n";
-#endif
-    // TODO: uncomment these lines, and implement diagnostics stuff
-    //compute_diagnostics = true;
-    //compute_energy      = true;
+  if (nstep_end%params.state_frequency==0 || nstep_end==tl.nstep0 ||
+      nstep_end>=last_time_step) {
+    compute_diagnostics = true;
   }
 
   if (params.disable_diagnostics) {
     compute_diagnostics = false;
   }
 
   if (compute_diagnostics) {
-    Errors::runtime_abort("'compute diagnostic' functionality not yet available in C++ build.\n",
-                          Errors::err_not_implemented);
-    // call prim_diag_scalars(elem,hvcoord,tl,4,.true.,nets,nete)
+    Diagnostics& diags = Context::singleton().get_diagnostics();
+    diags.prim_diag_scalars(true,3);
+    diags.prim_energy_halftimes(true,2);
   }
 
+  tl.update_tracers_levels(params.qsplit);
+
   // Apply forcing
 #ifdef CAM
-HybridVCoord  Errors::runtime_abort("CAM forcing not yet availble in C++.\n"
+  Errors::runtime_abort("CAM forcing not yet availble in C++.\n"
                         Errors::err_not_implemented);
   // call TimeLevel_Qdp(tl, qsplit, n0_qdp)
 
   // if (ftype==0) then
   //   call t_startf("ApplyCAMForcing")
-  //   call ApplyCAMForcing(elem, hvcoord,tl%n0,n0_qdp, dt_remap,nets,nete)
+  //   call ApplyCAMForcing(elem, fvm, hvcoord,tl%n0,n0_qdp, dt_remap,nets,nete)
```

Comment:
Also nbd, but fvm is not part of HOMMEXX anymore, so I'd prefer not to have this change